dimod/generators/mimo.py

```diff
@@ -0,0 +1,1085 @@
+                   attenuation_matrix: Optional[np.ndarray] = None) -> Tuple[
+                       np.ndarray, float, np.random.mtrand.RandomState]:
+    """Create a channel model.
+
+    Channel power is the expected root-mean-square signal per receiver (i.e.,
+    :math:`mean(F^2)*num_transmitters`) for homogeneous codes.
+
+    Args:
+        num_receivers: Number of receivers.
+
+        num_transmitters: Number of transmitters.
+
+        F_distribution: Distribution for the channel. Supported values are:
+
+        *   First value: ``normal`` and ``binary``.
+        *   Second value: ``real`` and ``complex``.
+
+        random_state: Seed for a random state or a random state.
+
+        attenuation_matrix: Root of the power associated with each 
+            transmitter-receiver channel. 
+
+    Returns:
+        Two-tuple of channel and channel power, where the channel is an 
+        :math:`i \times j` matrix with :math:`i` rows corresponding to the 
+        receivers and :math:`j` columns to the transmitters, and channel power 
+        is a number.
+    """
+
+    if num_receivers < 1 or num_transmitters < 1:
+        raise ValueError('At least one receiver and one transmitter are required.')
+    
+    channel_power = num_transmitters
+
+    random_state = _make_random_state(random_state)
+    
+    if F_distribution is None:
+        F_distribution = ('normal', 'complex')
+    elif type(F_distribution) is not tuple or len(F_distribution) != 2:
+        raise ValueError('F_distribution should be a tuple of strings or None')
+
+    if F_distribution[0] == 'normal':
+        if F_distribution[1] == 'real':
+            F = random_state.normal(0, 1, size=(num_receivers, num_transmitters))
+        else:
+            channel_power = 2 * num_transmitters
+            F = random_state.normal(0, 1, size=(num_receivers, num_transmitters)) + \
+                1j*random_state.normal(0, 1, size=(num_receivers, num_transmitters))
+    elif F_distribution[0] == 'binary':
+        if F_distribution[1] == 'real':
+            F = (1 - 2*random_state.randint(2, size=(num_receivers, num_transmitters)))
+        else:
+            channel_power = 2*num_transmitters      #For integer precision purposes:
+            F = ((1 - 2*random_state.randint(2, size=(num_receivers, num_transmitters))) +
+                 1j*(1 - 2*random_state.randint(2, size=(num_receivers, num_transmitters))))
+
+    if attenuation_matrix is not None:
+        if np.iscomplex(attenuation_matrix).any():
+            raise ValueError('attenuation_matrix must not have complex values')
+        F = F * attenuation_matrix    #Dense format for now, this is slow.
+        channel_power *= np.mean(np.sum(attenuation_matrix * attenuation_matrix,
+                                        axis=0)) / num_receivers
+
+    return F, channel_power
+    
+constellation = {   # bits per transmitter (bpt) and amplitudes (amps)
```

Comment:
I am not sure why this is defined here specifically. It should probably be at the top unless it's only used in one function.

Reply:
I think I forgot to delete this after adding the `mod_config` superset of modulation properties.

Reply:
Removed